pkg/kubelet/apis/config/v1beta1/defaults.go

```diff
@@ -286,4 +288,13 @@ func SetDefaults_KubeletConfiguration(obj *kubeletconfigv1beta1.KubeletConfigura
 	if obj.PodLogsDir == "" {
 		obj.PodLogsDir = DefaultPodLogsDir
 	}
+
+	if obj.SingleProcessOOMKill == nil && runtime.GOOS == "linux" {
```

Comment:
haven't we hit problems serializing defaults based on the current OS? won't this cause problems when kubeadm on a linux machine makes a kubelet configuration for a windows machine?

xref cmd/kubeadm/app/componentconfigs/kubelet_windows.go and bugs in the past like https://github.com/kubernetes/kubernetes/pull/123136/files

Reply:
round-tripping the config shouldn't produce different results depending on your OS

Reply:
Yes, that's more what I expected